include/boost/math/ccmath/sin.hpp

```diff
@@ -0,0 +1,98 @@
+//  (C) Copyright Matt Borland 2022.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_CCMATH_TRIG_HPP
+#define BOOST_MATH_CCMATH_TRIG_HPP
+
+#include <cmath>
+#include <type_traits>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/tools/is_constant_evaluated.hpp>
+#include <boost/math/tools/config.hpp>
+#include <boost/math/ccmath/abs.hpp>
+#include <boost/math/ccmath/isinf.hpp>
+#include <boost/math/ccmath/isnan.hpp>
+
+namespace boost::math::ccmath {
+
+namespace detail {
+
+template <typename T>
+[[nodiscard]] constexpr T sin_impl(T x) noexcept
+{
+    // Precalculated coeffs from the minmax polynomial for up to 128 bit long doubles
```

Comment:
Are your long doubles 128 bits?

Reply:
No but they have sufficient precision. -mlong-double-128 is not supported on x86_64 because it implicitly calls `__float128`. In order to support that I would have to make the end user link `-lquadmath` anytime these functions are used.